### ISSUE TITLE
Refactor main Umple project to make it more consistent (and better suited to a Gradle build)

### ISSUE THREAD

**Issue by AdamBJ** [umple/umple.gradle#18]
The convention we've specified for when the Gradle plugin is used in a Gradle build is that .ump files are located in `src/{source set name}/umple`. For example, if we're using the `main` `SourceSet`, the umple files should be in `src/main/umple`. The umple sub projects involved in a build violate this convention in various ways. Some umple files are stored in `{sub project root}/src/`, others are stored in `{source project root}/UmpleTLTemplates`. We should refactor the directory structure of the subprojects for consistency and to make it easier to build Umple using our Gradle plugin.

Another issue is that some project's `master.ump` tell the umple compiler to output generated files to a hard-coded location. For example, `UmpleParser`'s `master.ump` file tells Umple to output the Java files that are generated to `../src-gen-umple`. As long as we're still using the Ant build we need to keep these hard coded locations. They slow the Gradle build down a lot though because it means we have to make two copies of the generated source: one that gets stored at the hardcoded location, and one that is stored at the (correct) location we specify in the Gradle build.

**Comment by AdamBJ** [comment]
@Nava2 @vahdat-ab I'm going to do some of the directory restructuring work as part of the work I'm doing for the Gradle quick build. The hard coded stuff can't really be dealt with until we either change the Ant build (bad idea since we're going to replace it) or replace it with a full Gradle build, though. We shouldn't forget about getting rid of the hard coded junk once people have switched over to the Gradle build! If we get rid of it builds should be faster.

**Comment by Nava2** [comment]
> The hard coded stuff can't really be dealt with until we either change the Ant build (bad idea since we're going to replace it) or replace it with a full Gradle build, though. We shouldn't forget about getting rid of the hard coded junk once people have switched over to the Gradle build! If we get rid of it builds should be faster.

Don't forget, the main intention is to have both building in parallel for awhile while bugs are ironed out. This means changing the ant build. Don't need to change it much, but moving the folders around is reasonable to change within Ant.

**Comment by vahdat-ab** [comment]
don't do any changes to Ant files or its structures. We will have Gradle in the parallel with Ant. Do whatever is required for Gradle without breaking Ant :-)
